### DISCUSSION TITLE
Build strategy that is Dockerfile (for simple use cases)

### DISCUSSION THREAD

**Discussion by aslom** [knative/func#1623] [Prototype Ideas]
That was mentioned during today WG call. 

If I understand it correctly that would be simpler build strategy when buildpacks or s2i is not needed? @lance

**Comment by lance** [comment]
I think a reasonable option would be to add a `docker` package here: https://github.com/knative/func/tree/main/pkg/builders and when specified in `func.yaml` or on the command line as `func build --builder=docker` the CLI looks for a `Dockerfile` and builds using the docker/podman APIs. @matejvasek wdyt?

**Reply by zroubalik** [reply]
Are we going to provide Dockerfiles with the language templates or do we keep it on users?

**Reply by lance** [reply]
Good question! We could start with templates in knative-sandbox/func-tastic, perhaps?

**Comment by lkingland** [comment]
This is now a planned feature, and will likely be implemented as a so-called "Base Image".  The necessary Functions middleware is injected by `func` itself, so building with Buildpack, S2I, or directly on the host (outside of a container) are all very similar.  What is needed is a way to easily specify that the final runtime image be a given base image built from, for example, a Dockerfile; a feature currently only available via a custom buildpack with a Dockerfile for the "run" image.

Currently the plan is to make this available via the flag `--base-image=[myimage]`.  The base image itself could be built any way the user sees fit, either within the Function's build pipeline or externally.
